src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,264 @@ public static class NUnitTestExtractorApp
         {
-            if (!string.IsNullOrEmpty(includeInfo))
-            {
-                //Remove quotes from string in order to properly process it
-                includeInfo = includeInfo.Replace("\"", "");
+            HashSet<Type> testSuites = new HashSet<Type>();
 
-                List<string> includeInfoValues = new List<string>(includeInfo.Split(','));
-                
-                foreach (string value in includeInfoValues)
+            LoadAssembly(assemblyFullPath);
+
+            foreach (var item in _assembly.GetTypes())
+            {
+                if (IsValidTestSuite(item))
                 {
-                    if (categoryNames != null && !categoryNames.Contains(value))
-                    {
-                        return false;
-                    }
+                    testSuites.Add(item);
                 }
-                return true;
             }
-            throw new ArgumentNullException("includeInfo", "no includeInfo specified");
 
+            return testSuites;
         }
 
         /// <summary>
-        /// Used to determine whether or not a test/testcase should be excluded or not
+        /// Get all valid tests from the suite
         /// </summary>
-        /// <param name="categoryNames">List of categories to compare info to</param>
-        /// <param name="excludeInfo">string of info used to compare to categoryNames</param>
-        /// <returns>returns true if excludeInfo is inside of the categoryNames List, false otherwise</returns>
-        public static bool Exclude(IReadOnlyList<string> categoryNames, string excludeInfo)
+        /// <param name="testSuite"></param>
+        /// <returns>Return a list of tests</returns>
+        private static HashSet<string> GetValidTestsFromTestSuite(Type testSuite)
         {
-            if (!string.IsNullOrEmpty(excludeInfo))
+            HashSet<string> tests = new HashSet<string>();
+
+            foreach (MethodInfo test in testSuite.GetMethods())
             {
-                excludeInfo = excludeInfo.Replace("\"", "");
+                var attributes = test.GetCustomAttributesData();
+
+                // search function attributes to find out whether the function has "Test" or "TestCase" attribute defined
+                // if no matches, skip this function
+                // search function attributes to find out whether the function has "Explicit" or "Ingored" attribute defined
+                // if found any attribute related to "Explicit" or "Ingored", skip this function
+                if (!attributes.Any(x => x.AttributeType.Equals(typeof(TestAttribute))
+                                    && x.AttributeType.Equals(typeof(TestCaseAttribute)))
+                                    || attributes.Any(x => x.AttributeType.Equals(typeof(ExplicitAttribute)))
+                                    || attributes.Any(x => x.AttributeType.Equals(typeof(IgnoreAttribute))))
```

Comment:
[Must?] I don't think the logic is right here.  Shouldn't the && be || instead?
Won't the first 2 lines only return false if the test doesn't have both [Test] and [TestCase]?